tests/ReturnTypeTest.php

```diff
@@ -1264,6 +1264,29 @@ function produceParameters(array $configuration): array
                     }
                     ',
             ],
+            'NewFromTemplateObject' => [
+                'code' => '<?php
+                    /** @psalm-consistent-constructor */
+                    class AggregateResult {}
+
+                    /**
+                     * @template T as AggregateResult
+                     * @param T $type
+                     * @return T
+                     */
+                    function aggregate($type) {
+                        $t = new $type;
+                        return $t;
+                    }
+
+
+                    class b extends AggregateResult {}
+
+                    $b = aggregate(new b);',
```

Comment:
For clarity, the new b here (and even the whole function call) should not be part of the unit test given it already works as expected: https://psalm.dev/r/33cf9009d7

Stopping [InvalidReturnType](https://psalm.dev/011) is the sole purpose of the test I think